test/unit/instrumentation/http2/http2.test.js

```diff
@@ -0,0 +1,873 @@
+    const { agent, http2, protocol, host, port } = t.nr
+    const path = '/path'
+    helper.runInTransaction(agent, function () {
+      t.nr.transaction = agent.getTransaction()
+      makeRequest(
+        http2,
+        {
+          protocol,
+          host,
+          port,
+          path,
+          method: 'GET'
+        },
+        finish
+      )
+    })
+
+    function finish() {
+      const { transaction } = t.nr
+      const [, child] = transaction.trace.getChildren(transaction.trace.root.id)
+      const attributes = child.getAttributes()
+      assert.equal(attributes.url, `${protocol}://${host}:${port}${path}`)
+      assert.equal(attributes.procedure, 'GET')
+      end()
+    }
+  })
+
+  await t.test('should start and end segment', (t, end) => {
+    const { agent, http2, protocol, host, port } = t.nr
+    let segment
+    helper.runInTransaction(agent, async function () {
+      t.nr.transaction = agent.getTransaction()
+      segment = agent.tracer.getSegment()
+
+      assert.ok(segment.timer.hrstart instanceof Array)
+      assert.equal(segment.timer.hrDuration, null)
+      makeRequest(
+        http2,
+        {
+          protocol,
+          host,
+          port,
+          path: '/',
+          method: 'GET'
+        },
+        finish
+      )
+    })
+
+    async function finish() {
+      const { transaction } = t.nr
+      const [, child] = transaction.trace.getChildren(transaction.trace.root.id)
+
+      assert.ok(child.timer.hrDuration instanceof Array)
+      assert.ok(child.timer.getDurationInMillis() > 0)
+      transaction.end()
+      end()
+    }
+  })
+
+  await t.test('should not modify parent segment when parent segment opaque', (t, end) => {
+    const { agent, http2, protocol, host, port } = t.nr
+    const path = '/asdf?a=123&b=abcde'
+    helper.runInTransaction(agent, function (transaction) {
+      t.nr.transaction = transaction
+      const parentSegment = agent.tracer.createSegment({
+        name: 'ParentSegment',
+        parent: transaction.trace.root,
+        transaction
+      })
+      parentSegment.opaque = true
+      t.nr.parentSegment = parentSegment
+      agent.tracer.setSegment({ transaction, segment: parentSegment }) // make the current active segment
+      makeRequest(
+        http2,
+        {
+          protocol,
+          host,
+          port,
+          path,
+          method: 'GET'
+        },
+        finish
+      )
+    })
+
+    function finish() {
+      const segment = agent.tracer.getSegment()
+
+      assert.equal(segment, t.nr.parentSegment)
+      assert.equal(segment.name, 'ParentSegment')
+
+      const attributes = segment.getAttributes()
+      assert.ok(!attributes['request.parameters.a'])
+      assert.ok(!attributes['request.parameters.b'])
+      end()
+    }
+  })
+
+  await t.test('should not record segments and should not error if not run in a transaction', { skip: nodeVersion < 22 }, (t, end) => {
```

Comment:
this seems like copy pasta, it's not using raw headers and should run on all versions

Reply:
This test is in the outbound block, and defaults to object headers. Should each of these tests be run with object headers, and again with raw? (They certainly can be, if I functionally prepare the header block.)